src/main/java/javaslang/collection/Traversable.java

```diff
@@ -553,6 +540,77 @@ default T fold(T zero, BiFunction<? super T, ? super T, ? extends T> op) {
     Option<T> headOption();
 
     /**
+     * Returns the index of the first occurrence of the given element or -1 if this does not contain the given element.
+     *
+     * @param element an element
+     * @return the index of the first occurrence of the given element
+     */
+    default int indexOf(T element) {
+        return indexOf(element, 0);
+    }
+
+    /**
+     * Returns the index of the first occurrence of the given element after or at some start index
+     * or -1 if this does not contain the given element.
+     *
+     * @param element an element
+     * @param from start index
+     * @return the index of the first occurrence of the given element
+     */
+    int indexOf(T element, int from);
+
+    /**
+     * Finds first index where this sequence contains a given sequence as a slice.
+     * <p>
+     * Note: may not terminate for infinite-sized collections.
+     * @param that the sequence to test
+     * @return the first index such that the elements of this sequence starting at this index match
+     * the elements of sequence that, or -1 of no such subsequence exists.
+     * @throws NullPointerException if {@code that} is null.
+     */
+    default int indexOfSlice(Iterable<? extends T> that) {
+        Objects.requireNonNull(that, "that is null");
+        return indexOfSlice(that, 0);
+    }
+
+    /**
+     * Finds first index after or at a start index where this sequence contains a given sequence as a slice.
+     * <p>
+     * Note: may not terminate for infinite-sized collections.
+     * @param that the sequence to test
+     * @param from the start index
+     * @return the first index >= from such that the elements of this sequence starting at this index match
+     * the elements of sequence that, or -1 of no such subsequence exists.
+     * @throws NullPointerException if {@code that} is null.
+     */
+    default int indexOfSlice(Iterable<? extends T> that, int from) {
```

Comment:
This could even be Iterable<?> rather than Iterable<?>. Consider this scenario:
```java
List<CharSequences> that = Arrays.asList("a", "b");
Traversable<String> traversable = ..
traversable.indexOf(that, 0)
```
there could be a valid index for the given list `that`. On the other hand, it'll open the door for the same kind of bug-prone signatures as `java.util.List<T>.containsAll(Collection<?> c);`.